wagtail/models/__init__.py

```diff
@@ -1124,7 +1124,7 @@ def save(self, clean=True, user=None, log_action=False, **kwargs):
         such as updating the ``url_path`` value of descendant page to reflect changes
         to this page's slug.
 
-        New pages should generally be saved via the ``add_child()`` or ``add_sibling()``
+        New pages should generally be saved via the `add_child() <https://django-treebeard.readthedocs.io/en/latest/api.html#treebeard.models.Node.add_child>`_ or `add_sibling() <https://django-treebeard.readthedocs.io/en/latest/api.html#treebeard.models.Node.add_sibling>`_
```

Comment:
Not sure we need this either sorry, firstly these links go to the latest docs for that project and not the stable version. 

But mor importantly these are methods on Page which could be better referenced in the Page model reference page. 

Additionally this gets into a bit of a complex situation where we are recommending how to add pages programmatically, which is probably a gap in our docs but not sure on this as a good solution. 

https://docs.wagtail.org/en/stable/reference/pages/model_reference.html

Reply:
Yeah, @lb- Thanks so much for the detailed response. I think I should have logged an issue first before making the PR. Actually, I decided to make the contribution directly as it was suggested by @gasman during a conversation on Slack after a complaint was made by a user. Should I mention you in the conversation so you can find it?

Then, as regards the Treebeard additions I made, I did for a number of reasons:

- As you have rightly said, it is not obvious. Think about it this way: the Treebeard was mentioned at the beginning of the page, then `add_child()` and `add_siblings()` were mentioned further down on the page. I don't think anyone would be able to make a connection that the mentioned methods are from Treebeard. Except you are familiar with Treebeard, you won't know. And even if you are, you might think they are Page methods. In other words, except there is an explicit reference that points to where a method belong, it is difficult knowing. 
- Secondly, as I mention earlier, I made the changes due to suggestions from @gasman on the Slack channel. I didn't think to raise an issue.  But I will note to make an issue next time on similar ocassions.

Reply:
Yeah, if this looks unfit to add to the page, then a separate documentation to explain where Treebeard is used and how would be best—as you suggested.

Reply:
Yeah, "adding a page programmatically" is definitely a gap. I was building my first e-commerce store with Wagtail, and this issue propped up for me. I had to do a good search around and questioning. Hectic. A documentation would take away the stress.

 Then, a documentation gap keeps glaring at me: what is views.py rarely spoken about in the  documentation, especially given its importance in Django itself?

 For functionalities Wagtail does not support and developers have to exit the page model to handles some data, which might include writing views, then understanding how to integrate these views with the page model is very important.  For instance, I was trying to reverse render a template that is part of the page tree from a view, but I just couldn't figure how. I was blocked now.

Reply:
Then on the Django's user model case, I think it important mentioning it. While building my E-commerce app, I got to the point of implementing a user model 
I was at a cross road of deciding if I should define a custom one or use a user model by Wagtail. Naturally, I assume Wagtail would define its own custom model, so I thought to search around to see how robust it is and see if it can fit my purpose without me defining another one. The fact that the user model used by Wagtail was not mentioned was super confusing and delaying for me.

Reply:
So we do need a more explicit reference on this

Reply:
Hey @Temidayo32  thanks for the reply - I did not understand the context of this (maybe next time a bit more detail in the description about how you got to this point, a link to the Slack maybe could help).

In this specific case we may just wait for other thoughts from other core team on this, again, I am happy of others think this is a good addition here.

Reply:
Yeah, I am still talking about the Django user model. I meant perhaps we need a more explicit reference on it

Reply:
Thread discussing this: https://wagtailcms.slack.com/archives/C81FGJR2S/p1666788708237379

I see other links to treebard already use point to `latest`, so for consistency we should do the same here. However it’d likely be more correct to the [Materialized Path trees](https://django-treebeard.readthedocs.io/en/latest/mp_tree.html) docs since that’s what Wagtail pages inherit from (those docs then point to the general API docs).

```suggestion
New pages should generally be saved via the `add_child() <https://django-treebeard.readthedocs.io/en/latest/mp_tree.html#treebeard.mp_tree.MP_Node.add_child>`_ or `add_sibling() <https://django-treebeard.readthedocs.io/en/latest/mp_tree.html#treebeard.mp_tree.MP_Node.add_sibling>`_
```